AsyncDisplayKit/Details/ASBatchContext.m

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "ASBatchContext.h"
+
+typedef NS_ENUM(NSInteger, ASBatchContextState) {
+  ASBatchContextStateFetching,
+  ASBatchContextStateCancelled,
+  ASBatchContextStateCompleted
+};
+
+@interface ASBatchContext ()
+{
+  ASBatchContextState _state;
```

Comment:
we don't guard this with a lock.  ASTableView / ASCollectionView support editing & updates off-main and we don't specify thread-affinity for notifying ASDK code that the batch fetching completed.  will this go :boom:?